src/cljs/ataru/hakija/application_handlers.cljs

```diff
@@ -66,24 +74,24 @@
   :application/handle-submit-error
   (fn [cofx [_ response]]
     {:db (-> (update (:db cofx) :application dissoc :submit-status)
-             (assoc :error {:message "Tapahtui virhe " :detail response}))
-     :dispatch-later [{:ms 30000 :dispatch [:application/clear-error]}]}))
-
-(reg-event-db
-  :application/clear-error
-  (fn [db _]
-    (dissoc db :error)))
+             (assoc :error {:message "Tapahtui virhe " :detail response}))}))
 
 (reg-event-fx
-  :application/submit-form
+  :application/submit
   (fn [{:keys [db]} _]
-    {:db       (assoc-in db [:application :submit-status] :submitting)
+    (send-application db :submit)
+    {:db       (-> db (assoc-in [:application :submit-status] :submitting) (dissoc :error))
```

Comment:
Is the `dissoc` necessary? Is there a case where we'll end up in `:application/submit` with `:error` something else than `nil`?